### ISSUE TITLE
[CLOSED] ui: add peers to node search

### ISSUE THREAD

**Issue by arturo-aparicio** [opensource-repos-with-issues/consul#4]
<a href="https://github.com/LevelbossMike"><img src="https://avatars.githubusercontent.com/u/242299?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [LevelbossMike](https://github.com/LevelbossMike)**
_Monday Jul 25, 2022 at 09:10 GMT_
_Originally opened as https://github.com/hashicorp/consul/pull/13875_

----

### Description
Follow-up of https://github.com/hashicorp/consul/pull/13771 to make the tests execute on CI - couldn't find a way to do this in the old PR.

For discussion please follow the [old PR](https://github.com/hashicorp/consul/pull/13771). Nothing has changed - I'm just creating a new PR so that tests execute properly on CI.

__TLDR;__

This PR:
* makes nodes searchable by peer-name
* fixes peer-name showing up in services search although peers feature was turned off

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

----

_**[LevelbossMike](https://github.com/LevelbossMike)** included the following code: https://github.com/hashicorp/consul/pull/13875/commits_

**Comment by arturo-aparicio** [comment]
<a href="https://github.com/LevelbossMike"><img src="https://avatars.githubusercontent.com/u/242299?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [LevelbossMike](https://github.com/LevelbossMike)**
_Monday Jul 25, 2022 at 09:25 GMT_

----

@evrowe I can't approve myself - would be great to get a 👍 for this one again. No changes other than renaming the `dc.nodes`-controller has been done - it was called `DcServicesController` because I forgot to rename it after copy-pasting the controller file. This is the same PR as https://github.com/hashicorp/consul/pull/13771
